src/transaction_builder.cpp

```diff
@@ -90,19 +111,17 @@ void TransactionBuilder::SendChangeTo(libzcash::SaplingPaymentAddress changeAddr
     tChangeAddr = boost::none;
 }
 
-bool TransactionBuilder::SendChangeTo(CTxDestination& changeAddr)
+void TransactionBuilder::SendChangeTo(CTxDestination& changeAddr)
 {
     if (!IsValidDestination(changeAddr)) {
-        return false;
+        throw std::runtime_error("Invalid change address, not a valid taddr.");
```

Comment:
Do the calling functions need an explicit bool back? It seems odd we could just get away with returning void here

Reply:
This function is only called in tests and in `AsyncRPCOperation_sendmany::main_impl()`. Prior to this change the code in the latter where we used this was: 
`assert(builder_.SendChangeTo(changeAddr));`
So I think that the user experience will be better after this change. I believe `assert(...)` spits out a line number and other debugging information, and causes the program to crash (tbh I'm not actually sure how zcashd would respond to this) versus returning a failed async operation result with the above error to the user.

That being said, when we use this in z_sendmany I think we would have already thrown an exception if we were going to fail that assertion.